app/components/global-search-box.js

```diff
@@ -1,58 +1,70 @@
-import Component from '@ember/component';
-import { computed } from '@ember/object';
-import { reads } from '@ember/object/computed';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
 import { isBlank } from '@ember/utils';
 import { cleanQuery } from 'ilios-common/utils/query-utils';
-import { task, timeout } from 'ember-concurrency';
+import { restartableTask, timeout } from 'ember-concurrency';
 
 const DEBOUNCE_MS = 250;
 const MIN_INPUT = 3;
 
-export default Component.extend({
-  iliosConfig: service(),
-  iliosSearch: service('search'),
-  intl: service(),
+export default class GlobalSearchBox extends Component {
+  @service iliosConfig;
+  @service('search') iliosSearch;
+  @service intl;
 
-  autocompleteCache: null,
-  autocompleteSelectedQuery: null,
-  internalQuery: null,
-  query: null,
-  classNames: ['global-search-box'],
-  'data-test-global-search-box': true,
-  search() {},
+  @tracked autocompleteCache;
+  @tracked autocompleteSelectedQuery;
+  @tracked internalQuery;
 
-  hasResults: reads('results.length'),
-  results: reads('autocomplete.lastSuccessful.value'),
+  get hasResults() {
+    return !!this.results?.length;
+  }
 
-  computedQuery: computed('autocompleteSelectedQuery', 'query', 'internalQuery', function () {
+  get results() {
+    return this.autocomplete.lastSuccessful?.value;
+  }
+
+  get computedQuery() {
     if (typeof this.autocompleteSelectedQuery === 'string') {
       return this.autocompleteSelectedQuery;
     }
     if (typeof this.internalQuery === 'string') {
       return this.internalQuery;
     }
-    return this.query;
-  }),
+    return this.args.query;
+  }
 
-  init() {
-    this._super(...arguments);
+  constructor() {
+    super(...arguments);
```

Comment:
This isn't needed after moving to a class, autocompleteCache = [] can just be declared as the property.

Reply:
:heavy_check_mark: